cmake/SetupCompilerOptions.cmake

```diff
@@ -229,8 +229,18 @@ if( BLT_CXX_STD STREQUAL c++98 )
     set(CMAKE_CXX_STANDARD 98)
 elseif( BLT_CXX_STD STREQUAL c++11 )
     set(CMAKE_CXX_STANDARD 11)
+    if (COMPILER_FAMILY_IS_XL)
```

Comment:
This looks fine, it's clear you are adding flags for xl for the c++11 & c++14 case. 
If we need a fix for other cmakes thats ok as well, but our goal is to work with a few blessed versions of cmake -- currently 3.3.x and 3.8.x.